charts/vector/templates/keda.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if or (eq .Values.role "Aggregator") (eq .Values.role "Stateless-Aggregator") -}}
+{{- if and .Values.keda.enabled (not .Values.autoscaling.enabled)}}
```

Comment:
What do you think of raising a more explicit error if both are configured, like I've done here with [`fail`](https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/templates/_pod.tpl#L120)